mods/default/lua/environment/grow_bamboo.lua

```diff
@@ -0,0 +1,41 @@
+-- mods/default/lua/environment/grow_bamboo.lua
+-- =============================================
+-- See README.txt for licensing and other information.
+
+-- Bamboo
+function default.grow_bamboo(pos, node)
+	pos.y = pos.y - 1
+	local name = core.get_node(pos).name
+	if name ~= "default:dirt_with_grass" and name ~= "default:dirt" then
+		return
+	end
+	pos.y = pos.y + 1
+	local height = 0
+	while node.name == "default:bamboo" and height < 15 do
+		height = height + 1
+		pos.y = pos.y + 1
+		node = core.get_node(pos)
+	end
+	if height > 10 or node.name ~= "air" then
```

Comment:
so ... the nodes will be setted if the height is larger than 10 __or__ the node (above?) is not air.

So when the node above is e.g. stone and the height is e.g. 3 the nodes also will be setted. Maybe you want to use `and` not `or`.